internal/digraph/executor/docker.go

```diff
@@ -108,14 +109,17 @@ func (e *docker) Run(ctx context.Context) error {
 		args = append(args, val)
 	}
 
-	// If containerName is set, use exec instead of creating a new container
-	if e.containerName != "" {
+	// If image is not set or execConfig is set, use exec instead of creating a new container
+	if e.image == "" || e.execConfig != nil {
+		if e.containerName == "" {
+			return fmt.Errorf("containerName must be set in exec mode")
+		}
 		return e.execInContainer(ctx, cli, args)
 	}
 
 	// New container creation logic
 	if e.pull {
-		reader, err := cli.ImagePull(ctx, e.image, image.PullOptions{})
+		reader, err := cli.ImagePull(ctx, e.image, image.PullOptions{Platform: e.platform})
```

Comment:
We could make this a little more robust by checking if the locally available image targets the correct platform or not and attempt to pull the correct one. WDYT @yottahmd ?

Reply:
Yes, I think this is a great idea. Could we do that?